GVRf/Framework/framework/src/main/jni/vulkan/vulkanCore.cpp

```diff
@@ -922,14 +923,24 @@ void VulkanCore::InitPipelineForRenderData(const GVR_VK_Vertices* m_vertices, Vu
             getVKSampleBit(sampleCount), VK_NULL_HANDLE, VK_NULL_HANDLE, VK_NULL_HANDLE,
             VK_NULL_HANDLE, VK_NULL_HANDLE);
 
-    pipelineCreateInfo.pDepthStencilState = gvr::PipelineDepthStencilStateCreateInfo(rdata->depth_test() ? VK_TRUE : VK_FALSE,
-                                                                                     rdata->depth_mask() ? VK_TRUE : VK_FALSE,
-                                                                                     VK_COMPARE_OP_LESS_OR_EQUAL,
-                                                                                     VK_FALSE,
-                                                                                     VK_STENCIL_OP_KEEP,
-                                                                                     VK_STENCIL_OP_KEEP,
-                                                                                     VK_COMPARE_OP_ALWAYS,
-                                                                                     VK_FALSE);
+
+    bool depthWrite = (rdata->rendering_order() == RenderData::Queue::Stencil) ? false : true;
+
+    pipelineCreateInfo.pDepthStencilState =
+            gvr::PipelineDepthStencilStateCreateInfo(rdata->depth_test() ? VK_TRUE : VK_FALSE,
+                                     (rdata->depth_mask()  && depthWrite)? VK_TRUE : VK_FALSE,
+                                     VK_COMPARE_OP_LESS_OR_EQUAL,
+                                     VK_FALSE,
+                                     static_cast<VkStencilOp>(m_conversionFlags.getFromGL(rdata->stencil_op_sfail())),  //stencil pass
+                                     static_cast<VkStencilOp>(m_conversionFlags.getFromGL(rdata->stencil_op_dppass())), //depth pass, stencil pass
+                                     static_cast<VkStencilOp>(m_conversionFlags.getFromGL(rdata->stencil_op_dpfail())), //depth fail, stencil pass
+                                     static_cast<VkCompareOp>(m_conversionFlags.getFromGL(rdata->stencil_func_func())), //compare function
+                                     rdata->stencil_func_mask(), //compare mask
+                                     rdata->getStencilMask(), //stencil mask
```

Comment:
do this only if stencil flag is enabled.

Reply:
Should this be done in the GL renderer too then? We are not checking if the stencil formats are supported before we setup stencil testing in setRenderStates.